File: 2021/6/6.py
from collections import Counter
fish = [int(x) for x in open("6.in").read().strip().split(",")]
fish_grouped = Counter(fish)
for day in range(256):
    new_fish = fish_grouped[0]
    for i in range(1,9):
        fish_grouped[i-1] = fish_grouped[i]
    fish_grouped[6] += new_fish
    fish_grouped[8] = new_fish
    if day == 79 or day == 255:
        print(f"Flock size after {day+1} days:", sum(fish_grouped.values()))

File: 2021/6/6.rb
fish = File.read("6.in").chomp.split(",").map(&:to_i)
fish_grouped = fish.tally
fish_grouped.default = 0
256.times do |day|
  new_fish = fish_grouped[0]
  1.upto(8) do |i|
    fish_grouped[i-1] = fish_grouped[i]
  end
  fish_grouped[6] += new_fish
  fish_grouped[8] = new_fish
  if day == 79 || day == 255
    puts "Flock size after #{day+1} days: #{fish_grouped.values.sum}"
  end
end
